### PR TITLE
HeaderedControls

### PR DESCRIPTION
**Pull request by niels9001**
- Creating a new `HeaderedControls` component that includes the `HeaderedContentControl`, `HeaderedItemsControl` and `HeaderedTreeView`.
- Ported changes for `HeaderedItemsControl` from: https://github.com/CommunityToolkit/Labs-Windows/pull/418
- Ported `HeaderedTreeView` from: https://github.com/CommunityToolkit/Labs-Windows/pull/418
- Added sample / .md for the `HeaderedTreeView`
- Removed the earlier ported `HeaderedContentControl` and `HeaderedItemsControl`

### CODE REVIEW

File: components/HeaderedControls/tests/HeaderedControls.Tests.projitems
Comment (michael-hawker) [inline components/HeaderedControls/tests/HeaderedControls.Tests.projitems:1]: Probably be good to at least have a basic test case for each control which loads it on a page and checks that it can be grabbed, means that if we accidentally break anything with logic/loading/template that we may catch it?

**Comment by niels9001** [comment]
@michael-hawker I think I addressed your feedback :-). Added the same tests that are normally generated by default.

File: components/HeaderedControls/tests/HeaderedTreeViewTestClass.cs
Comment (michael-hawker) [inline components/HeaderedControls/tests/HeaderedTreeViewTestClass.cs:42]: We can remove some of these trivial tests probably? Or you thinking of leaving them for future examples for writing new tests here later?

File: components/HeaderedControls/samples/Assets/Sunny.png
Comment (michael-hawker) [inline components/HeaderedControls/samples/Assets/Sunny.png:1]: Hmm, this isn't showing up on the UWP sample... (Was also having trouble seeing the code samples in the sample app). Maybe something for us to investigate outside the PR? Or is this a faux-png image that's causing issues maybe? Worked in WASDK no problem though...

File: components/HeaderedControls/samples/HeaderedTreeViewSample.xaml
Comment (michael-hawker) [inline components/HeaderedControls/samples/HeaderedTreeViewSample.xaml:1]: Hmm, this was crashing in UWP for me...

"Specified cast is not valid"/"No such interface supported"

Worked fine in WASDK...

I haven't seen issues in my DataTable usage (think I've tested both)...

This smells like a case where a WUX version of TreeView or TreeViewItem is trying to be used somewhere? I couldn't see off-hand where that might be though... Ah... just found it...

File: components/HeaderedControls/samples/HeaderedTreeViewSample.xaml.cs
Comment (michael-hawker) [inline components/HeaderedControls/samples/HeaderedTreeViewSample.xaml.cs:108]: Brace formatting is weird here. +Private variable is just `_children`? https://learn.microsoft.com/en-us/dotnet/csharp/fundamentals/coding-style/coding-conventions#camel-case

Thought we had editorconfig/style warnings for this stuff...? @Arlodotexe?

File: components/HeaderedControls/samples/HeaderedTreeViewSample.xaml
Comment (michael-hawker) [inline components/HeaderedControls/samples/HeaderedTreeViewSample.xaml:31]:
```suggestion
                    <muxc:TreeViewItem Content="{x:Bind Name}"
                                  IsExpanded="True"
                                  ItemsSource="{x:Bind Children}" />
```

This by default is a WUX TreeViewItem in UWP, which doesn't jive... needs to be MUXC TreeViewItem in a MUXC TreeView, missing namespace inclusion at the top should have been a giveaway too 😋

**Review by michael-hawker — CHANGES_REQUESTED** [review]
Couple minor fixes needed, otherwise looks good. I'll push a fix for the WUX/MUXC issue though quick as I tested locally.

File: components/HeaderedControls/samples/HeaderedTreeViewSample.xaml.cs
Comment (niels9001) [reply components/HeaderedControls/samples/HeaderedTreeViewSample.xaml.cs]: Sloppy copy-and-paste from the WinUI Gallery sample :). Fixed now!

File: components/HeaderedControls/samples/Assets/Sunny.png
Comment (Arlodotexe) [reply components/HeaderedControls/samples/Assets/Sunny.png:1]: Seeing the same issue on my machine for the UWP head.

**Review by michael-hawker — APPROVED** [review]

File: components/HeaderedControls/samples/Assets/Sunny.png
Comment (michael-hawker) [reply components/HeaderedControls/samples/Assets/Sunny.png:1]: @Arlodotexe want to open an issue? Not sure how best to describe this problem, but don't think would be unique to this PR, so don't want to hold this up.

File: components/HeaderedControls/src/CommunityToolkit.WinUI.Controls.HeaderedControls.csproj
Comment (Arlodotexe) [inline components/HeaderedControls/src/CommunityToolkit.WinUI.Controls.HeaderedControls.csproj:12]: Need to fix the PackageId before merging

**Review by Arlodotexe — CHANGES_REQUESTED** [review]

**Review by Arlodotexe — APPROVED** [review]

**Comment by michael-hawker** [comment]
Thanks @Arlodotexe for fixing that!

**Pull request merged**
